src/lib/Preview.js

```diff
@@ -1226,6 +1246,27 @@ class Preview extends EventEmitter {
         this.viewer.load(err);
     }
 
+    /**
+     * Message, to any listeners of Preview, that an error has occurred.
+     *
+     * @private
+     * @param {Error} error - The error that occurred.
+     * @return {void}
+     */
+    logPreviewError(error) {
+        const err = error;
+        // Make sure to strip auth
+        err.message = stripAuthFromString(error.message || '');
+        err.displayMessage = stripAuthFromString(error.displayMessage || '');
+
+        const errorLog = {
+            error: err,
+            data: error.data || ''
```

Comment:
How would we access the error type for analytics/debugging? Would that be in the error.message? Would that potentially overwrite a message we would get such as `cannot access property hasTouch of undefined`?

Reply:
For logging purposes, we specify our own error codes (see events.js). If we have not thrown the error ourselves, setting the message to one of our error codes, the `error.message` will be that of what the browser created. 

I'm thinking a single logged event will be as follows:
```
{
  event_type: <error | metric>
  logger_session_id: <string> // ID of session 
  timestamp: <string>, // ISO Format
  file_id: <string>,
  file_version_id: <string>,
  content_type: <string>, // Content type of the file previewed
  extension: <string>,
  client_version: <string>, // preview version number
  browser_name: <string>,
  locale: <string>,                        
  event_name: <string>, // Either our error code OR browser error message
  value: <any> // Must be serializable
}
```

Note: I just realized I need to update so that we're sending more session related info, per `prevew_error` event.

Reply:
Should we separate that out so that we have
`error.code`: our custom code we made or a generic one if the browser threw it
`error.messge`: what the browser created / more info from if required

If all of that info will be needed by whatever is posting our logs, don't we need to emit that stuff in the error events? Or do we emit that once per preview and we expect our consumer to retain that information for us.